datasets/msr_sqa/msr_sqa.py

```diff
@@ -0,0 +1,113 @@
+tables from Wikipedia. We had three workers decompose each WTQ question, resulting in a dataset of 6,066 sequences \
+that contain 17,553 questions in total. Each question is also associated with answers in the form of cell \
+locations in the tables.
+"""
+
+_HOMEPAGE = "https://msropendata.com/datasets/b25190ed-0f59-47b1-9211-5962858142c2"
+
+_LICENSE = "Microsoft Research Data License Agreement"
+
+_URL = "https://download.microsoft.com/download/1/D/C/1DC270D2-1B53-4A61-A2E3-88AB3E4E6E1F/SQA%20Release%201.0.zip"
+
+
+class MsrSQA(datasets.GeneratorBasedBuilder):
+    """Microsoft Research Sequential Question Answering (SQA) Dataset"""
+
+    def _info(self):
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "annotator": datasets.Value("int32"),
+                    "position": datasets.Value("int32"),
+                    "question": datasets.Value("string"),
+                    "table_file": datasets.Value("string"),
+                    "table_header": datasets.features.Sequence(datasets.Value("string")),
+                    "table_data": datasets.features.Sequence(datasets.features.Sequence(datasets.Value("string"))),
+                    "answer_coordinates": datasets.features.Sequence(datasets.Value("string")),
+                    "answer_text": datasets.features.Sequence(datasets.Value("string")),
+                }
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = os.path.join(dl_manager.download_and_extract(_URL), "SQA Release 1.0")
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"filepath": os.path.join(data_dir, "train.tsv"), "data_dir": data_dir},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={"filepath": os.path.join(data_dir, "test.tsv"), "data_dir": data_dir},
+            ),
+        ]
+
+    def _generate_examples(self, filepath, data_dir):
+        """ Yields examples. """
+
+        def load_table_data(table_file):
+            with open(table_file, encoding="utf-8") as f:
+                lines = f.readlines()
+            header = lines[0].strip().split(",")
+            data = [line.strip().split(",") for line in lines[1:]]
+            return header, data
+
+        with open(filepath, encoding="utf-8") as f:
+            reader = csv.DictReader(f, delimiter="\t")
+            for row in reader:
+                item = dict(row)
+                item["answer_text"] = eval(item["answer_text"])
+                item["answer_coordinates"] = eval(item["answer_coordinates"])
```

Comment:
Could you parse their values without using `eval` ?
`eval` is considered unsafe

Reply:
Perhaps you can use the methods I proposed [here](https://github.com/huggingface/datasets/issues/880#issuecomment-732939750), which are used in the original implementation of TAPAS by Google AI (which achieves SOTA on SQA).

Reply:
@NielsRogge Thanks for the suggestion 💡